### PR TITLE
Feature add more complex methods repositories

### PR DESCRIPTION
**Pull request by pachecowolox**
## Summary
- add findByPublisherAndGenreAndYear in BookController
- add findByBirthdateBetweenAndNameIgnoreCase in UserController
## Trello Card
https://trello.com/c/9OWMbMKw/25-agregar-m%C3%A9todos-m%C3%A1s-complejos-a-los-repositorios

### CODE REVIEW

File: src/main/java/wolox/training/repositories/UserRepository.java [hunk 1]
```diff
@@ -8,4 +10,6 @@
 
 	Optional<User> findByUsername(String username);
 
+	List<User> findByBirthdateBetweenAndNameIgnoreCase(LocalDate startDate, LocalDate endDate,String name);
```
Comment (juanwongramirez) [inline src/main/java/wolox/training/repositories/UserRepository.java]: According to the card the attribute name must contain a character sequence and not be equals, you can replace the method to **findByBirthdateBetweenAndNameContainingIgnoreCase**

**Review by juanwongramirez — CHANGES_REQUESTED** [review]

File: src/main/java/wolox/training/repositories/UserRepository.java (hunk 1 above)
Comment (pachecowolox) [reply src/main/java/wolox/training/repositories/UserRepository.java]: done

**Review by juanwongramirez — DISMISSED** [review]
In the future it might to be convenient add unit tests for any changes and new methods

File: src/main/java/wolox/training/controllers/BookController.java [hunk 2]
```diff
@@ -143,4 +143,21 @@ public Book updateBook(@RequestBody Book book, @PathVariable Long id) {
...
+	 *
+	 * @param publisher: publisher of the book (Object)
+	 * @param genre:     genre of the book (Object)
+	 * @param year:      year of the book (Object)
+	 *
+	 * @return {@link Book}
+	 */
+
+	@GetMapping("/parameters")
+	public List<Book> findByPublisherAndGenreAndYear(@RequestParam(required = false) String publisher,
```
Comment (marcegl91) [inline src/main/java/wolox/training/controllers/BookController.java:158]: Nice! The Trello card does not ask you to provide and endpoint but is nice that you have added it!

File: src/main/java/wolox/training/repositories/BookRepository.java [hunk 3]
```diff
@@ -9,4 +12,9 @@
 	Optional<Book> findByAuthor(String author);
 
 	Optional<Book> findByIsbn(String author);
+
+	@Query("SELECT b FROM Book b WHERE (:publisher is null or b.publisher = :publisher) and (:genre is null or b"
+			+ ".genre = :genre) and (:year is null or b.year = :year)")
+	List<Book> findByPublisherAndGenreAndYear(@Param(value = "publisher") String publisher,
```
Comment (marcegl91) [inline src/main/java/wolox/training/repositories/BookRepository.java:19]: Why do you need this `@Param` annotation that is from Feign? You are not using Feign here.

**Review by marcegl91 — CHANGES_REQUESTED** [review]

**Review by marcegl91 — APPROVED** [review]
Great!

**Review by juanwongramirez — APPROVED** [review]

**Pull request merged**